src/components/Groups/GroupPage/FeedItem.tsx

```diff
@@ -77,11 +92,116 @@ export default function FeedItem({
   isBeingDragged?: boolean;
 }): JSX.Element {
   const { showAdminView } = useActiveGroup();
+  const firebaseApp = useFirebaseApp();
   const { updatePost, deletePost } = usePostActions(group.id);
 
   const [showDropdown, setShowDropdown] = useState(false);
+  const [showExportModal, setShowExportModal] = useState(false);
+
+  const { firebaseUser, isLoaded } = useContext(UserDataContext);
+  const groups = useUserGroups();
+  const [problems, setProblems] = React.useState<GroupProblemData[]>([]);
+
   const ref = React.useRef<HTMLDivElement>();
 
+  const map = new Map<string, MapData>();
+
+  function exportSelectedPosts() {
+    const type = post.type;
+    const defaultPost: Omit<PostData, 'timestamp'> = {
+      name: post.name,
+      isPublished: post.isPublished,
+      isPinned: post.isPinned,
+      body: post.body,
+      isDeleted: post.isDeleted,
+      type,
+      pointsPerProblem: post.pointsPerProblem,
+      problemOrdering: post.problemOrdering,
+      ...(type === 'announcement'
+        ? {}
+        : {
+            dueTimestamp: null,
+          }),
+    };
+
+    const q = query(
+      collection(
+        getFirestore(firebaseApp),
+        'groups',
+        group.id,
+        'posts',
+        post.id,
+        'problems'
+      ) as CollectionReference<GroupProblemData>
+    );
+
+    onSnapshot(q, {
+      next: snap => {
+        setProblems(snap.docs.map(doc => ({ id: doc.id, ...doc.data() })));
+      },
+      error: error => {
+        toast.error(error.message);
+      },
+    });
+
+    map.forEach(async (value: MapData, key: string) => {
+      if (value.used) {
+        const firestore = getFirestore(firebaseApp);
+        const batch = writeBatch(firestore);
+        const docRef = doc(
+          collection(getFirestore(firebaseApp), 'groups', key, 'posts')
+        );
+
+        batch.set(docRef, { ...defaultPost, timestamp: serverTimestamp() });
+        batch.update(doc(firestore, 'groups', key), {
+          postOrdering: arrayUnion(docRef.id),
+        });
+        await batch.commit();
+        // if(post.type != 'announcement') {
+        const docRef2 = doc(
+          collection(
+            getFirestore(firebaseApp),
+            'groups',
+            key,
+            'posts',
+            docRef.id,
+            'problems'
+          )
+        );
+
+        problems.map(async problem => {
```

Comment:
problems are not exported... maybe you need to await this (also wrap in `Promise.all`?)

Reply:
Forgot to add that, I'll fix it.